hypershift-operator/controllers/nodepool/metrics/metrics.go

```diff
@@ -193,11 +202,80 @@ func createFailureConditionToNodePoolsCountMap(knownConditionToExpectedStatus ma
 	return &res
 }
 
+const (
+	unexpectedAWSOutputErrorReason          = "unexpected AWS output"
+	failedToCallAWSErrorReason              = "failed to call AWS"
+	unableToMarshalPricingDataErrorReason   = "unable to marshal pricing data"
+	unableToUnMarshalPricingDataErrorReason = "unable to un-marshal pricing data"
+	noErroReason                            = ""
+)
+
 type vCpusDetail struct {
```

Comment:
Suggestion / nice to have / nitpick:
I know I created this type but I think that it is in fact not that useful. We could probably remove it in favour of having `retrieveVCpusDetailsPerNode` returns an `int32` (the `vCpusCount`) and an `errorReason`; `errorReason` would be a new type wrapping a `string` and implementing the `error` interface.
`extractCPUFromInstanceTypeName` signature could be changed to have the same return types than `retrieveVCpusDetailsPerNode`.
WDYT?